src/universal/modules/welcome/containers/Welcome/Welcome.js

```diff
@@ -32,16 +49,27 @@ const mapStateToProps = state => ({
 export default class WelcomeContainer extends Component {
   static propTypes = {
     dispatch: PropTypes.func,
+    invitees: PropTypes.array,
+    inviteesRaw: PropTypes.string,
+    preferredName: PropTypes.string,
+    teamName: PropTypes.string,
     welcome: PropTypes.shape({
-      preferredName: PropTypes.string,
```

Comment:
Since the form isn't destroyed now until it's abandoned, the form data will still be in redux. So, you'll still get that benefit :)